File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        int n = deck.size();
        sort(deck.begin(), deck.end());
        vector<int> ans(n);
        queue<int> pq;

        for(int i = 0; i<n; i++) pq.push(i);

        for(int i = 0; i<n; i++) {
            ans[pq.front()] = deck[i];
            pq.pop();
            pq.push(pq.front());
            pq.pop();
        }

        return ans;
    }
};

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.java
class Solution(object):
    def deckRevealedIncreasing(self, deck):
        n = len(deck)
        deck.sort()
        pq = collections.deque(range(n))
        ans = [0] * n

        for i in range(n):
            ans[pq.popleft()] = deck[i]
            if pq:
                pq.append(pq.popleft())

        return ans
        